### PR TITLE
[Docs] More detailed M1 Mac installation instructions

### PR DESCRIPTION
**Pull request by wuisawesome**
## Why are these changes needed?

This PR adds more detail the M1 mac installation instructions following the bug bash.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/installation.rst [hunk 1]
```diff
@@ -159,14 +159,22 @@ The latest Ray Java snapshot can be found in `sonatype repository <https://oss.s
...
-Apple Silicon Support
+M1 Mac support (Apple Silicon) Support
 ---------------------
 
 Ray has experimental support for machines running Apple Silicon (such as M1 macs). To get started:
 
 1. Install `miniforge <https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh>`_.
-2. Ensure that the ``grpcio`` package is installed via forge and **not pypi**: ``pip uninstall grpcio; conda install grpcio``.
-3. Install Ray as you normally would: ``pip install ray``.
+  a. wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh
```
Comment (pcmoritz) [inline doc/source/installation.rst:168]: These should be formatted in a monospace font (e.g. with ``)

File: doc/source/installation.rst [hunk 2]
```diff
@@ -159,14 +159,22 @@ The latest Ray Java snapshot can be found in `sonatype repository <https://oss.s
 
 .. _apple-silcon-supprt:
 
-Apple Silicon Support
+M1 Mac support (Apple Silicon) Support
```
Comment (gjoliver) [inline doc/source/installation.rst:162]: get rid of the first "support"?

File: doc/source/installation.rst [hunk 3]
```diff
@@ -159,14 +159,22 @@ The latest Ray Java snapshot can be found in `sonatype repository <https://oss.s
...
 
 1. Install `miniforge <https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh>`_.
-2. Ensure that the ``grpcio`` package is installed via forge and **not pypi**: ``pip uninstall grpcio; conda install grpcio``.
-3. Install Ray as you normally would: ``pip install ray``.
+  a. wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh
+  b. bash Miniforge3-MacOSX-arm64.sh
+  c. rm https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh # Cleanup.
+2. Ensure you're using the miniforge environment (you should see (base) in your terminal).
+  a. ``source ~/.bash_profile``
+  b. ``conda activate``
```
Comment (gjoliver) [inline doc/source/installation.rst:173]: do you not need the name of the env?

**Review by gjoliver — COMMENTED** [review]
quick comments.

File: doc/source/installation.rst (hunk 3 above)
Comment (wuisawesome) [reply doc/source/installation.rst:173]: It defaults to `base`

**Review by gjoliver — APPROVED** [review]

**Comment by richardliaw** [comment]
Lint is failing here. @wuisawesome please be sure to check lint before merge.

**Pull request merged**